hypervisor/include/arch/x86/timer.h

```diff
@@ -7,36 +7,60 @@
 #ifndef TIMER_H
 #define TIMER_H
 
+/**
+ * @brief Timer
+ *
+ * @defgroup timer ACRN Timer
+ * @{
+ */
+
 typedef void (*timer_handle_t)(void *data);
 
+/**
+ * @brief Definition of timer tick mode
+ */
 enum tick_mode {
-	TICK_MODE_ONESHOT = 0,
-	TICK_MODE_PERIODIC,
+	TICK_MODE_ONESHOT = 0,	/**< one-shot mode */
+	TICK_MODE_PERIODIC,	/**< periodic mode */
 };
 
+/**
+ * @brief Definition of timers for per-cpu
+ */
 struct per_cpu_timers {
-	struct list_head timer_list;	/* it's for runtime active timer list */
+	struct list_head timer_list;	/**< it's for runtime active timer list */
 };
 
+/**
+ * @brief Definition of timer
+ */
 struct hv_timer {
-	struct list_head node;		/* link all timers */
-	enum tick_mode mode;			/* timer mode: one-shot or periodic */
-	uint64_t fire_tsc;		/* tsc deadline to interrupt */
-	uint64_t period_in_cycle;	/* period of the periodic timer in unit of TSC cycles */
-	timer_handle_t func;		/* callback if time reached */
-	void *priv_data;		/* func private data */
+	struct list_head node;		/**< link all timers */
+	enum tick_mode mode;		/**< timer mode: one-shot or periodic */
+	uint64_t fire_tsc;		/**< tsc deadline to interrupt */
+	uint64_t period_in_cycle;	/**< period of the periodic timer in unit of TSC cycles */
+	timer_handle_t func;		/**< callback if time reached */
+	void *priv_data;		/**< func private data */
 };
 
-/*
- * Don't initialize a timer twice if it has been add to the timer list
- * after call add_timer. If u want, delete the timer from the list first.
+/* External Interfaces */
+
+/**
+ * @brief Initialize a timer structure.
+ *
+ * @param[in] timer Pointer to timer.
+ * @param[in] func irq callback if time reached.
+ * @param[in] priv_data func private data.
+ * @param[in] fire_tsc tsc deadline to interrupt.
+ * @param[in] mode timer mode.
+ * @param[in] period_in_cycle period of the periodic timer in unit of TSC cycles.
+ *
+ * @remark Don't initialize a timer twice if it has been add to the timer list
```

Comment:
I believe the preference is to have a `@return` directive even for functions that do not return any value (specifically: `@return None`). @junjiemao1 can you chime in?

Reply:
Exactly as @gvancuts said, please use ''@return None'' to make the absence of a return value explicit.